### PR TITLE
Clean up the widget pattern in the static-pages code

### PR DESCRIPTION
**Pull request by Rleahy22**
## Description
Refactored static-pages-entry and its corresponding directory to make it easier to understand.

- [x] Convert the "social share links" to a typical widget
- [x] Convert the "alerts-dismiss-view" to a typical widget
- [x] Convert the "ics-generator" to a typical widget
- [x] Convert the "createMyVALoginWidget" to a typical widget
- [x] Consider restructuring the directory to reflect "widget creators" instead of having them spread out across various directories
- [x] Make sure all `data-widget-type` values are stored in the constant file, `widgetTypes.js`
- [x] Consider moving the CTA widget code from `platform/site-wide` into `static
- [ ] There is a lot of repetition of registering widgets in `static-pages-entry`. We should consider whether we can create an iterator to enforce future widgets follow the correct pattern.

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/7751

## Testing done
Local, unit, E2E

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by ashagross — APPROVED** [review]
LGTM!

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: src/applications/static-pages/widget-creators/alerts-dismiss-view.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/**
```
Comment (kelsonic) [inline src/applications/static-pages/widget-creators/alerts-dismiss-view.js:1]: This was copy-pasted from the old alerts-dismiss-view, right? 🙂

File: src/applications/static-pages/widgetTypes.js [hunk 2]
```diff
@@ -9,18 +11,31 @@ export default {
   DISABILITY_APP_STATUS: 'disability-app-status',
   DISABILITY_RATING_CALCULATOR: 'disability-rating-calculator',
   EDUCATION_APP_STATUS: 'education-app-status',
+  FACILITY_APPOINTMENT_WAIT_TIMES_WIDGET:
+    'facility-appointment-wait-times-widget',
+  FACILITY_DETAIL: 'facility-detail',
+  FACILITY_LOCATIONS_LIST: 'facility-locations-list',
+  FACILITY_MAP: 'facility-map',
+  FACILITY_PATIENT_SATISFACTION_SCORES: 'facility-patient-satisfaction-scores',
```
Comment (kelsonic) [inline src/applications/static-pages/widgetTypes.js:19]: Wow this is awesome 💯

**Review by kelsonic — APPROVED** [review]
Thank you!! 🔥

File: src/applications/static-pages/widget-creators/alerts-dismiss-view.js (hunk 1 above)
Comment (Rleahy22) [reply src/applications/static-pages/widget-creators/alerts-dismiss-view.js:1]: I actually moved it and made (small) changes but apparently git missed it. It doesn't attach the event listener immediately anymore and instead checks for the existence of the anchor elements before running the code

**Review by timwright12 — APPROVED** [review]

**Review by mpelzsherman — APPROVED** [review]

**Pull request merged**